library/core/src/ptr/non_null.rs

```diff
@@ -1803,4 +1813,46 @@ mod verify {
         let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    #[kani::proof_for_contract(NonNull::as_ptr)]
+    pub fn non_null_check_as_ptr() {
+        // Create a non-null pointer to a random value
+        let non_null_ptr: *mut i32 = kani::any::<usize>() as *mut i32;
+        if let Some(ptr) = NonNull::new(non_null_ptr){
+        // Call as_ptr
+            let raw_ptr = ptr.as_ptr();}
+    
+    }
+    #[kani::proof_for_contract(NonNull::<[T]>::as_mut_ptr)]
+    pub fn non_null_check_as_mut_ptr() {
+        // Create a non-null slice pointer
+        const ARR_LEN: usize = 100;
+        let mut generator = kani::PointerGenerator::<ARR_LEN>::new();
+        let mut values: [i32; ARR_LEN] = [0; ARR_LEN];
+        let raw_ptr: *mut i32 = generator.any_in_bounds().ptr;
```

Comment:
Please use `any_alloc_status()` instead. Or the arbitrary implementation from https://github.com/model-checking/verify-rust-std/pull/128/

Reply:
Thanks, I have used it in as_mut_ptr and as_non_null_ptr